### PR TITLE
add remote-setup in plcontainer management tool

### PR DESCRIPTION
**Pull request by Sasasu**
```
./management/bin/plcontainer remote-setup -h
usage: plcontainer remote-setup [-h] --hosts HOSTS [--clientdir CLIENTDIR]

optional arguments:
  -h, --help            show this help message and exit
  --hosts HOSTS         a list of remote host IP (e.g. '128.66.0.1,128.66.0.2') or a gpscp hostfile
  --clientdir CLIENTDIR
                        the dir to save plcontainer language client
```

### CODE REVIEW

**Comment by Sasasu** [comment]
ci https://dev2.ci.gpdb.pivotal.io/teams/gp-extensions/pipelines/dev.plcontainer.sa_removedockertools.sa

File: management/bin/plcontainer [hunk 1]
```diff
@@ -152,17 +161,37 @@ def distribute_docker_file(pool, d_file, dest, hosts):
...
     pool.join()
     pool.check_results()
 
+def distribute_clientdir(pool, dest, hosts):
+    clientdir = os.path.join(DBHOME, PLCONTAINER_CLIENT_DIR)
+    if _SCP:
+        # `scp   -r foo/bin/plcontainer_clients dest_dir` will copy plcontainer_clients/* to dest_dir
+        # `rsync -r foo/bin/plcontainer_clients dest_dir` will copy plcontainer_clients   to dest_dir
+        # in scp mode, copy the parent dir instead, this will also copy foo/bin/plcontainer into dest_dir.
+        clientdir = os.path.dirname(clientdir)
```
Comment (RMTT) [inline management/bin/plcontainer:176]: Can use `PLCONTAINER_CLIENT_DIR/*` for both?

File: management/bin/plcontainer [hunk 2]
```diff
@@ -226,7 +257,8 @@ def install_image_remote(hosts, pool, docker_image):
             logger.error("Cannot load docker image in all hosts")
         elif (step == 3):
             logger.error("Unable to remove temporary files")
-        raise(ex)
+        raise ex
```
Comment (RMTT) [inline management/bin/plcontainer:260]: What the difference between `raise ex` and `raise(ex)`

Comment (Sasasu) [reply management/bin/plcontainer:252]: no different.
raise is a keyword not a function. `()` equals to nothing

```

In [7]: ast.dump(ast.parse('raise 1'))
Out[7]: 'Module(body=[Raise(exc=Constant(value=1))], type_ignores=[])'

In [8]: ast.dump(ast.parse('raise()'))
Out[8]: 'Module(body=[Raise(exc=Tuple(elts=[], ctx=Load()))], type_ignores=[])'

In [9]: ast.dump(ast.parse('raise(1)'))
Out[9]: 'Module(body=[Raise(exc=Constant(value=1))], type_ignores=[])'

In [10]: ast.dump(ast.parse('raise(1,)'))
Out[10]: 'Module(body=[Raise(exc=Tuple(elts=[Constant(value=1)], ctx=Load()))], type_ignores=[])'
```

**Comment by Sasasu** [comment]
one squashed commit. 6878ae1d58fbeb2307b75e0dbc87f421bfa6852e
use `PLCONTAINER_CLIENT_DIR/*` for both rsync and scp

File: tests/regress/expected/test_configuration.out [hunk 3]
```diff
@@ -1,3 +1,11 @@
+-- start_ignore
+\! docker ps -a
```
Comment (RMTT) [inline tests/regress/expected/test_configuration.out:2]: i think the output is unnecessary.

File: tests/regress/expected/test_configuration.out [hunk 4]
```diff
@@ -19,6 +27,14 @@ SELECT pyconf();
      10
 (1 row)
 
+-- start_ignore
```
Comment (RMTT) [inline tests/regress/expected/test_configuration.out:30]: i think the output is unnecessary.

**Review by RMTT — CHANGES_REQUESTED** [review]

File: tests/regress/expected/test_configuration.out (hunk 4 above)
Comment (Sasasu) [reply tests/regress/expected/test_configuration.out:30]: this is some debug message to explain why `count(*) from plcontainer_containers_summary()`  returns `1`.
keep this message can show what happened in the design time. in this case, there are many running container but only one is belong to dbid=self.

and this file is copied from `tests/regress/result/`. remove those lines need to edit the output by hand.

File: tests/regress/expected/test_configuration.out (hunk 3 above)
Comment (Sasasu) [reply tests/regress/expected/test_configuration.out:2]: this is some debug message to explain why count(*) from plcontainer_containers_summary() returns 1.
keep this message can show what happened in the design time. in this case, there are many running container but only one is belong to dbid=self.

and this file is copied from tests/regress/result/. remove those lines need to edit the output by hand.

**Review by RMTT — APPROVED** [review]

**Pull request merged**